packages/colorpickers/src/utils/saturation.ts

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import { IHSVColor } from '../utils/types';
+
+export function limit(value: number, max: number, min = 0) {
+  if (value < min) {
+    return min;
+  } else if (value > max) {
+    return max;
+  }
+
+  return value;
+}
+
+const getSaturationLightness = (element: HTMLDivElement, x: number, y: number, rtl: boolean) => {
+  const { width, height } = element.getBoundingClientRect();
+
+  let left = x - (element.getBoundingClientRect().left + window.pageXOffset);
+  let top = y - (element.getBoundingClientRect().top + window.pageYOffset);
+
+  if (left < 0) {
+    left = 0;
+  } else if (left > width) {
+    left = width;
+  }
+
+  if (top < 0) {
+    top = 0;
+  } else if (top > height) {
+    top = height;
+  }
+
+  if (rtl) {
+    left = width - left;
+  }
+
+  const saturation = left / width;
+  const bright = 1 - top / height;
+
+  return {
+    saturation,
+    bright
+  };
+};
+
+export function getNextHsv(
+  e: MouseEvent,
+  hue: number,
+  container: HTMLDivElement,
+  rtl: boolean
+): IHSVColor {
+  const { saturation, bright } = getSaturationLightness(container, e.pageX, e.pageY, rtl);
+
+  return {
+    h: hue,
+    s: saturation,
+    v: bright
+  };
+}
+
+export const getThumbPosition = (
```

Comment:
Interesting bugs, the first bug is a result of the hue slider having an incorrect max hue of `359`. It should be `360`. The color values (hex, rgb) should be calculated correctly now.

The second bug seems to be related to Storybook rather than the color picker. It can be replicated for other color pickers such as [Fluent UI color picker](https://fluent-color-spike.netlify.app/?path=/docs/components-colorpicker--default) and [react-color](https://color-preview.netlify.app/?path=/docs/components-colorpicker-colorpicker--default). I'll investigate if there's something that can be done at the Storybook level to address this.